packages/sanity/src/ui-components/button/Button.tsx

```diff
@@ -23,6 +23,7 @@ type BaseButtonProps = Pick<
   | 'tone'
   | 'type'
   | 'width'
+  | 'radius'
```

Comment:
I think we should not include extra props supported in the ui-components. Can we handle this in other way from the consumer? Do we strictly need this change?

Reply:
Radius is already being passed through to the UIButton. This isn't changing the behaviour more just ensuring the type is more accurate. Perhaps we should update this button to pass through only the props we `Pick` here, however I worry as to where might be knock on impact we don't intend

Reply:
It will be passed and I think it's ok but you will get a TS warning saying the prop is not supported.
If the dev want's to bypass that he could add a `ts-ignore` comment.
But we want to enforce consistency in the buttons used inside the Studio, that's why we are only allowing some props.

Reply:
I am of the opinion that if we are to be strict, we should be really strict. eg right now I've evaded some cumbersome restrictions by just passing a `style` prop through to `ui-component`'s button. Really, if we are opinionated on what the studio button is then, besides a small number of props, we shouldn't just extend all on the sanity/ui's Button.

However I appreciate your point regarding `radius` here, so have reverted this